### ISSUE TITLE
Curious about apply_topn and apply_greedy

### ISSUE THREAD

**Issue by littlefive5** [hannamw/EAP-IG#8]
Can you provide the differences between these two ways to get the subgraph and which is better?

Thanks

**Comment by hannamw** [comment]
Hi! The first way just selects the top *n* edges by their score. It's easy to run on any model and even large graphs. 

Apply greedy explores the graph starting from the logits node. At every timestep from 1...*n*, it chooses the highest-scoring edge connected to the current circuit (again, this is initialized to the logits node alone), and adds its parent to the circuit if it's not in the circuit yet. This tends to lead to better-connected circuits, which means that after pruning away all nodes with no parents / no children (and corresponding edges), we end up with a number of edges that's closer to *n*. 

Unfortunately, doing this sort of graph exploration ended up kind of expensive the way I initially implemented it (using a heap). So, I can't recommend this option for larger models.

**Comment by hannamw** [comment]
Feel free to ask or email with any more questions!
